serviceregistration/kubernetes/client/config.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import (
+	"crypto/x509"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/hashicorp/vault/sdk/helper/certutil"
+)
+
+const (
+	// These environment variables aren't set by default.
+	// Vault may read them in if set through these environment variables.
```

Comment:
Maybe we should add a comment about order of operations (IE are config values preferred over envs)?

Reply:
Oh yes, good idea! Will do.

Reply:
https://github.com/hashicorp/vault/pull/8249/commits/e75fe3b5a0e18f954173bd37242a5e5a7c92c586